Create Language-Version-History.md

```diff
@@ -0,0 +1,110 @@
+
+- Interactive environment
+- Object programming
+- Encapsulation Extensions
+
+# [F# 2.0](https://fsharp.org/specs/language-spec/2.0/FSharpSpec-2.0-April-2012.pdf)
+
+- Active patterns
+- Units of measure
+- Object expressions
+- Sequence expressions
+- Asynchronous programming
+- Agent programming
+- Extension members
+- Named arguments
+- Optional arguments
+- Array slicing
+- Quotations
+- Native interoperability
+- Computation expressions
+
+# [F# 3.0](https://fsharp.org/specs/language-spec/3.0/FSharpSpec-3.0-final.pdf)
+
+- Type providers
+- LINQ query expressions
+- CLIMutable attribute
+- Triple-quoted strings
+- Auto-properties
+- Provided units-of-measure
+
+# [F# 3.1](https://fsharp.org/specs/language-spec/3.1/FSharpSpec-3.1-final.pdf)
+
+- Named union type fields
+- Extensions to array slicing
+- Type inference enhancements
+
+# [F# 4.0](https://fsharp.org/specs/language-spec/4.0/FSharpSpec-4.0-final.pdf)
+
+- `printf` on unitized values
+- Extension property initializers
+- Non-null provided types
+- Primary constructors as functions
+- Static parameters for provided methods
+- `printf` interpolation
+- Extended `#if` grammar
+- Tailcall attribute
+- Multiple interface instantiations
+- Optional type args
+- Params dictionaries
+
+# [F# 4.1](https://fsharp.org/specs/language-spec/4.1/FSharpSpec-4.1-latest.pdf)
+
+- Struct tuples which inter-operate with C# tuples
+- Struct annotations for Records
+- Struct annotations for Single-case Discriminated Unions
+- Underscores in numeric literals
+- Caller info argument attributes
+- Result type and some basic Result functions
+- Mutually referential types and modules within the same file
+- Implicit `Module` syntax on modules with shared name as type
+- Byref returns, supporting consuming C# `ref`-returning methods
+- Error message improvements
+- Support for `fixed`
+
+# [F# 4.5](https://docs.microsoft.com/en-us/dotnet/fsharp/whats-new/fsharp-45)
+
+- Versioning alignment of binary, package, and language
+- Support for `Span<'T>` and related types
+- Ability to produce `byref` returns
+- The `voidptr` type
+- The `inref<'T>` and `outref<'T>` types to represent readonly and write-only `byref`s
+- `IsByRefLike` structs
+- `IsReadOnly` structs
+- Extension method support for `byref<'T>`/`inref<'T>`/`outref<'T>`
+- `match!` keyword in computation expressions
+- Relaxed upcast with `yield` in F# sequence/list/array expressions
+- Relaxed indentation with list and array expressions
+- Enumeration cases emitted as public
+
+# [F# 4.6](https://docs.microsoft.com/en-us/dotnet/fsharp/whats-new/fsharp-46)
+
+- Anonymous records
+- `ValueOption` module functions
+
+# [F# 4.7](https://docs.microsoft.com/en-us/dotnet/fsharp/whats-new/fsharp-47)
+
+- compiler support for `LangVersion`
+- Implicit `yield`s
+- No more required double underscore (wildcard identifier)
+- Indentation relaxations for parameters passed to constructors and static methods
+- `nameof` function (preview)
```

Comment:
Should the preview bits be their own section?

Reply:
I'd suggest they not be mentioned at all until they've landed in a 'stable' release. Eg `nameof` would not be mentioned until 5.0 (presuming it's promoted to stable at that point, naturally).

Reply:
@baronfel, could you expand on your reasoning on not mentionning those?

I assume promoting the existence of preview features allow more eyeballs / usage of those, which would be good side effect.

Some thoughts coming from pointing about preview:

* I agree with singling them out as proposed by @Happypig375 under a separate list under the language version rather than _(preview)_ suffix
* when a new version is released which ships them, we remove those from that list in the preview version, and mention as suffix _(was preview from version X)_

This will keep things non redundant as time progresses, while also separating preview from baked features visually, achieving the intent about promotion and better visual cues / not mixed with features that don't require the "langversion:preview" flag which may trip end users if they dismiss the small visual cue.

@Happypig375 did you intend the file to be named Language-Version-History.md rather than the current name?

Reply:
The preview section also would mean that we have an equivalent of the Feature column of the [Language Feature Status of C#.](https://github.com/dotnet/roslyn/blob/master/docs/Language%20Feature%20Status.md)